app/javascript/application.js

```diff
@@ -1 +1,4 @@
 import './controllers';
+import "bootstrap";
+import "@popperjs/core";
+
```

Comment:
Optimizing bundle size by loading only the Bootstrap plugins we need is usually a good idea, but for a project this complex, it might be safer to just import the full library. This way, we ensure everything works consistently across different parts of the project, even if other contributors add components later on. It also helps avoid any issues with missing plugins or functionality popping up unexpectedly.